src/peft/tuners/lora/layer.py

```diff
@@ -188,6 +188,18 @@ def reset_lora_parameters(self, adapter_name, init_lora_weights):
                 nn.init.kaiming_uniform_(self.lora_A[adapter_name].weight, a=math.sqrt(5))
             elif init_lora_weights.lower() == "gaussian":
                 nn.init.normal_(self.lora_A[adapter_name].weight, std=1 / self.r[adapter_name])
+            elif init_lora_weights.lower() == "orthogonal":
+                with torch.no_grad():
+                    rank = self.r[adapter_name]
+                    X = torch.randn(rank, rank)
+                    Q, _ = torch.linalg.qr(X)
+                    set1 = Q[0::2,:]  # Odd rows
+                    set2 = Q[1::2,:]  # Even rows
```

Comment:
`r` needs to be even for this to work, right? Let's check it and raise an error with a helpful message if it's not.